### DISCUSSION TITLE
asset massDelete functionality eats a lot of memory and crashes

### DISCUSSION THREAD

**Discussion by zssvaidar** [atrocore/atrocore#998] [Q&A]
nginx -> php-error
2023/10/31 17:10:29 [error] 771#771: *187354 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Allowed memory size of 2147483648 bytes exhausted (tried to allocate 20480 bytes) in /var/www/pim/vendor/pim/core/app/Espo/ORM/DB/Mapper.php on line 122" while reading response header from upstream, client: , server: , request: "DELETE /api/v1/Product/productAssets/relation HTTP/1.1", upstream: "fastcgi://unix:/var/run/php/php8.1-fpm.sock:", host: "",

After try to massDelete assets from product page with 1-2 assets attached to product, php eats up havely server memory, and then dies. Is this request becomes heavy load because we have 2million pictures in our system*? Have you tried optimizing this piece of code in next versions than our? Note: problem occur locally and on server also

[Delete] http://localhost/api/v1/Product/productAssets/relation 
body {
{
    "where": [
        {
            "type": "equals",
            "attribute": "id",
            "value": "649da92a3cc563cb2"
        }
    ],
    "foreignWhere": []
}}

modules versions we have been using:
pim 1.9.17
core 1.6.13
dam 1.5.14

Thanks you for support

**Comment by yurdm** [comment]
In newer versions, we reworked this code and that problem must be solved. Try to update your system to the latest version.
